src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentMap.java

```diff
@@ -20,31 +20,31 @@
  */
 public class WeakConcurrentMap<K, V> extends ReferenceQueue<K> implements Runnable, Iterable<Map.Entry<K, V>> {
 
-    private static final AtomicLong ID = new AtomicLong();
-
-    final ConcurrentMap<WeakKey<K>, V> target;
-
-    private final Thread thread;
-
     /**
      * Latent keys are cached thread-locally to avoid allocations on lookups.
-     * This is beneficial as the JIT unfortunately can't reliably replace the {@link LatentKey} allocation with stack allocations,
-     * even though the {@link LatentKey} does not escape.
+     * This is beneficial as the JIT unfortunately can't reliably replace the {@link LatentKey} allocation
+     * with stack allocations, even though the {@link LatentKey} does not escape.
      */
-    private static final ThreadLocal<LatentKey<?>> latentKeyThreadLocal = new ThreadLocal<LatentKey<?>>() {
+    private static final ThreadLocal<LatentKey<?>> LATENT_KEY_CACHE = new ThreadLocal<LatentKey<?>>() {
         @Override
         protected LatentKey<?> initialValue() {
             return new LatentKey<Object>();
         }
     };
 
+    private static final AtomicLong ID = new AtomicLong();
+
+    final ConcurrentMap<WeakKey<K>, V> target;
+
+    private final Thread thread;
+
     private final boolean reuseKeys;
 
     /**
      * @param cleanerThread {@code true} if a thread should be started that removes stale entries.
      */
     public WeakConcurrentMap(boolean cleanerThread) {
-        this(cleanerThread, !canBeUnloaded(WeakConcurrentMap.class.getClassLoader()));
+        this(cleanerThread, isPersistentClassLoader(LatentKey.class.getClassLoader()));
```

Comment:
This may throw a SecurityException. Do you think the getClassLoader call should be made in the try block of isPersistentClassLoader?

Reply:
This should never throw a security exception as any class is allowed to find class loaders in its own package, or did I overlook something?

Reply:
Ah, you are right. I missed that part in the Javadoc:
```
@throws  SecurityException
if a security manager is present, and the caller's class loader
is not {@code null} and is not the same as or an ancestor of the
class loader for the class whose class loader is requested,
and the caller does not have the
{@link RuntimePermission}{@code ("getClassLoader")}
```

Reply:
Yes, just verified it myself in the code if I remembered it right. The javadoc is not overly specific, though.